### PR TITLE
Add isGenerated, isPopulated, remove, and regenerate to Chunk.

### PR DESCRIPTION
**Pull request by Bammerbom**
https://github.com/SpongePowered/SpongeAPI/issues/360

### CODE REVIEW

File: src/main/java/org/spongepowered/api/world/Chunk.java [hunk 1]
```diff
@@ -71,4 +78,18 @@
...
+    
+    /**
+     * Removes the chunk.
+     *
+     * @return If the chunk was successfully removed
+     */
+    boolean remove();
+    
+    /**
+     * Reegenerates the chunk.
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/world/Chunk.java]: Typo?

Comment (Bammerbom) [inline src/main/java/org/spongepowered/api/world/Chunk.java]: Oops, fixed.

**Comment by gabizou** [comment]
Unless I am mistaken, but a `Chunk` is an instance of a chunk that has already passed through generation. During generation, in the API, we use [`MutableBlockBuffer`](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/world/extent/MutableBlockBuffer.java) to prepare the data that will later be transferred to a chunk. There should never exist a case where a `Chunk` has not been generated already.

Additionally, there already exists [`World.deleteChunk()`](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/world/World.java#L82) to remove a chunk. The method should not exist in the `Chunk` itself since it can not control whether it belongs to a world or not.

Regarding `regenerate()`, you can simply delete the chunk and [load the chunk](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/world/World.java#L72) again.

**Comment by Bammerbom** [comment]
And how about the isGenerated?

**Comment by Deamon5550** [comment]
>There should never exist a case where a Chunk has not been generated already.

**Comment by Bammerbom** [comment]
I mean, is there already a method for that

**Comment by gabizou** [comment]
@Bammerbom If you have an instance of a Chunk, it's already __*generated*__, so why would you need a method to check if a *generated* chunk `isGenerated`?

**Comment by Bammerbom** [comment]
I mean, is there any other method outside Chunk that does this yet?

**Comment by Deamon5550** [comment]
```java
if(world.loadChunk(Vector3i, false) == Optional.absent())
{
  //do shit
}
```

**Comment by Bammerbom** [comment]
Maybe add a better method to do this. But this pull request seems useless so someone can deny it.

**Comment by Bammerbom** [comment]
Should I create a pull request that adds a isGenerated method?

**Comment by gabizou** [comment]
> Should I create a pull request that adds a isGenerated method?

No. World#loadChunk#isPresent should be sufficient.

**Pull request closed without merging**